AnkiDroid/src/test/java/com/ichi2/anki/ModelFieldEditorTest.java

```diff
@@ -0,0 +1,155 @@
+package com.ichi2.anki;
+
+import android.content.Intent;
+import android.widget.EditText;
+
+import com.afollestad.materialdialogs.MaterialDialog;
+import com.ichi2.anki.exception.ConfirmModSchemaException;
+import com.ichi2.libanki.Model;
+import com.ichi2.libanki.Models;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import static com.afollestad.materialdialogs.DialogAction.POSITIVE;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(AndroidJUnit4.class)
+public class ModelFieldEditorTest extends RobolectricTest {
+    /**
+     * Tests if field names with illegal characters get removed from beginning of field names when adding field
+     */
+    @Test
+    public void testIllegalCharactersInFieldName_addField() {
+        // iterate through the forbidden characters and test
+        for (String forbidden : new String[] {"#", "^", "/", " ", "\t"}) {
+            String fieldName = setupInvalidFieldName(forbidden, true);
+
+            testForIllegalCharacters(fieldName);
+        }
+    }
+
+
+    /**
+     * Tests if field names with illegal characters get removed from beginning of field names when renaming field
+     */
+    @Test
+    public void testIllegalCharactersInFieldName_renameField() {
+
+        // iterate through the forbidden characters and test
+        for (String forbidden : new String[] {"#", "^", "/", " ", "\t"}) {
+            String fieldName = setupInvalidFieldName(forbidden, false);
+
+            testForIllegalCharacters(fieldName);
+        }
+    }
+
+
+    /**
+     * Assert that model's fields doesn't contain the forbidden field name
+     *
+     * @param forbiddenFieldName The forbidden field name to identify
+     */
+    private void testForIllegalCharacters(String forbiddenFieldName) {
+        List<String> modelFields = getCurrentDatabaseModelCopy("Basic").getFieldsNames();
+        String fieldName = getCurrentDatabaseModelCopy("Basic").getFieldsNames()
+                .get(modelFields.size() - 1);
```

Comment:
I think this was your intention
```suggestion
        String fieldName = modelFields.get(modelFields.size() - 1);
```